spark/src/main/scala/org/apache/spark/Plugins.scala

```diff
@@ -62,7 +62,13 @@ class CometDriverPlugin extends DriverPlugin with Logging with ShimCometDriverPl
         Math.max((executorMemory * memoryOverheadFactor).toLong, memoryOverheadMinMib)
       }
 
-      val cometMemOverhead = CometSparkSessionExtensions.getCometMemoryOverheadInMiB(sc.getConf)
+      val cometMemOverhead =
+        if (!CometSparkSessionExtensions.cometUnifiedMemoryManagerEnabled(sc.getConf)) {
+          CometSparkSessionExtensions.getCometMemoryOverheadInMiB(sc.getConf)
+        } else {
+          // comet shuffle unified memory manager is disabled, so we need to add overhead memory
+          CometSparkSessionExtensions.getCometShuffleMemorySize(sc.getConf)
```

Comment:
What is the reason behind we still need use `getCometShuffleMemorySize` instead of `0` when unified memory is disabled?

Reply:
Since the previous shouldOverrideMemoryConf is true, this if branch means cometShuffleUnifiedMemoryManagerInTestEnabled is false. So we need to reserve some overhead memory for shuffle allocator that do not use unified memory manager.

Reply:
Ah got it, thank you